hypershift-operator/controllers/hostedcluster/hostedcluster_controller.go

```diff
@@ -404,7 +404,13 @@ func (r *HostedClusterReconciler) Reconcile(ctx context.Context, req ctrl.Reques
 
 	// Reconcile the hosted cluster namespace
 	controlPlaneNamespace := manifests.HostedControlPlaneNamespace(hcluster.Namespace, hcluster.Name)
-	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, controlPlaneNamespace, NoopReconcile)
+	_, err = controllerutil.CreateOrUpdate(ctx, r.Client, controlPlaneNamespace, func() error {
+		if controlPlaneNamespace.Labels == nil {
+			controlPlaneNamespace.Labels = make(map[string]string)
+		}
+		controlPlaneNamespace.Labels[hyperv1.ControlPlaneComponent] = "namespace"
```

Comment:
err..Why? That this is a namespace and used for a hostedcluster is implicit or not...?

Reply:
Yes, that this is a hypershift control plane namespace. When `kube_namespace_labels` is fixed, we can adjust the recording rule to join on that instead of `kube_pod_labels` (and then we technically wouldn't need this label on our pods at all)

Reply:
Changed to hypershift.openshift.io/hosted-control-plane for now